### PR TITLE
[Fix] 401 redirection due to error in API calls

### PR DESCRIPTION
**Pull request by taylorz**
### Summary 

After updating much of the data fetching to useSWR some of the `put` api calls, [most](https://app.highlight.io/5415/sessions/b7S0aCD6QcmMS3XP5fR1XzG4B2J4?network-resource-id=7&page=1&query=and%7C%7Ccustom_processed%2Cis%2Ctrue) [notably](https://app.highlight.io/5415/sessions/m5cpPyxS47FEWp8YhWnUuDFxbyCq?network-resource-id=42&page=1&query=and%7C%7Ccustom_processed%2Cis%2Ctrue) those in `api/steps` would erroneously log the user out when an trying to assign a task. 

This PR moves the definition of `config` directly into each individual API call rather than globally doing so in `base` and also ensures that all `put` calls are passed an empty object for params so that the call doesn't fail.

### CODE REVIEW

File: api/schools.js [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
 }
 
 export const showSchool = {
   key: (schoolId, params) => `/v1/schools/${schoolId}`,
   fetcher: (schoolId, params) => {
+    const config = getAuthHeader();
+    config.params = params;
     return schoolsApi
-      .get(`/${schoolId}`, { params: params })
+      .get(`/${schoolId}`, config)
```
Comment (styliii) [inline api/schools.js:27]: Have you tested this? Just want to make sure the params are being passed in as expected.

File: pages/ssj/[workflow]/[phase]/[milestone]/index.js [hunk 2]
```diff
@@ -2,14 +2,14 @@ import { useRouter } from "next/router";
...
-import { Container, Draggable } from "react-smooth-dnd";
+// import { Container, Draggable } from "react-smooth-dnd";
 import { arrayMoveImmutable } from "array-move";
 import getAuthHeader from "@lib/getAuthHeader";
 import processesApi from "@api/workflow/processes";
 import { clearLoggedInState, redirectLoginProps } from "@lib/handleLogout";
 import Skeleton from "@mui/material/Skeleton";
 
-import useAuth from "@lib/utils/useAuth";
+// import useAuth from "@lib/utils/useAuth";
```
Comment (styliii) [inline pages/ssj/[workflow]/[phase]/[milestone]/index.js]: Are you sure you want to keep this commented out?

**Comment by styliii** [comment]
I think `loginEmailLink` needs a config being passed in.

What do you think of completely getting rid of 
```
  if (token) {
    // console.log("token from cookies", jwt_decode(token).sub)
    config.headers["Authorization"] = token;
  } else if (axios.defaults.headers.common["Authorization"]) {
    // console.log("token set from axios global", jwt_decode(axios.defaults.headers.common['Authorization']).sub);
  }
```
in `base.js`? This would prove to us that all the functions that need a auth header passed in was caught.

File: api/people.js [hunk 3]
```diff
@@ -32,6 +40,7 @@ async function update(personId, personParams) {
 }
 
 // filter example: {{ops_guide: true, rgl: true}}
+// TODO update with SWR hook
```
Comment (styliii) [inline api/people.js:45]: Do we need to pass in a config into this function?

File: api/schools.js (hunk 1 above)
Comment (taylorz) [reply api/schools.js:27]: I do see the params being passed in the call when I navigate to the schoolId page!

File: pages/ssj/[workflow]/[phase]/[milestone]/index.js (hunk 2 above)
Comment (taylorz) [reply pages/ssj/[workflow]/[phase]/[milestone]/index.js]: THanks for catching, do not want to keep commented

File: api/people.js (hunk 3 above)
Comment (taylorz) [reply api/people.js:45]: Yes! Adjusted

**Comment by taylorz** [comment]
> I think `loginEmailLink` needs a config being passed in.
> 
> What do you think of completely getting rid of
> 
> ```
>   if (token) {
>     // console.log("token from cookies", jwt_decode(token).sub)
>     config.headers["Authorization"] = token;
>   } else if (axios.defaults.headers.common["Authorization"]) {
>     // console.log("token set from axios global", jwt_decode(axios.defaults.headers.common['Authorization']).sub);
>   }
> ```
> 
> in `base.js`? This would prove to us that all the functions that need a auth header passed in was caught.

Added config to `loginEmailLink` thanks for catching-- and updated the people api call with config, thank you!
Also fully remove the code in question in api/base

**Pull request merged**